sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasBuilder.cs

```diff
@@ -324,75 +327,85 @@ private void EnsureState()
             }
             if (String.IsNullOrEmpty(this.Version))
             {
-                this.Version = SasQueryParameters.SasVersion;
+                this.Version = SasQueryParameters.DefaultSasVersion;
             }
         }
 
+        /// <summary>
+        /// Returns a string that represents the current object.
+        /// </summary>
+        /// <returns>A string that represents the current object.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override string ToString() =>
+            base.ToString();
+
         /// <summary>
         /// Check if two BlobSasBuilder instances are equal.
         /// </summary>
         /// <param name="obj">The instance to compare to.</param>
         /// <returns>True if they're equal, false otherwise.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public override bool Equals(object obj)
             => obj is BlobSasBuilder other && this.Equals(other);
 
         /// <summary>
         /// Get a hash code for the BlobSasBuilder.
         /// </summary>
         /// <returns>Hash code for the BlobSasBuilder.</returns>
-        public override int GetHashCode()
-            => this.BlobName.GetHashCode()
-            ^ this.CacheControl.GetHashCode()
-            ^ this.ContainerName.GetHashCode()
-            ^ this.ContentDisposition.GetHashCode()
-            ^ this.ContentEncoding.GetHashCode()
-            ^ this.ContentLanguage.GetHashCode()
-            ^ this.ContentType.GetHashCode()
-            ^ this.ExpiryTime.GetHashCode()
-            ^ this.Identifier.GetHashCode()
-            ^ this.IPRange.GetHashCode()
-            ^ this.Permissions.GetHashCode()
-            ^ this.Protocol.GetHashCode()
-            ^ this.StartTime.GetHashCode()
-            ^ this.Version.GetHashCode()
-            ;
+        [EditorBrowsable(EditorBrowsableState.Never)]
```

Comment:
You should be using the HashCodeBuilder (in Azure.Core shared source)

Reply:
I'll file a bug to update this everywhere in Storage.  It probably won't happen ASAP.

Reply:
Filed #6735